### PR TITLE
[DT] Mark partial slices unsupported in padding encoding resolver.

### PR DESCRIPTION
**Pull request by hanhanW**
The implementation was wrong; the tests are missing. It addresses a post review comment: https://github.com/iree-org/iree/pull/22339/files#r2438738373

### CODE REVIEW

**Review by kuhar — APPROVED** [review]
Looks good but could we add a test?

**Comment by hanhanW** [comment]
> Looks good but could we add a test?

We can only add negative test, but we don't test all corner cases. This PR should be treated as a post-fix of the other PR.

File: compiler/src/iree/compiler/Codegen/ExternalInterfaces/GPUEncodingExternalModels.cpp [hunk 1]
```diff
@@ -586,6 +586,14 @@ struct GPUPadEncodingLayoutMaterializerAttr final
     if (!boundType || !boundType.getEncoding()) {
       return failure();
     }
+    // Only handle cases where the slice spans the whole
+    // `!iree_tensor_ext.dispatch.tensor` type.
+    // TODO(hanchung): Enable partial slices. It was copied from pattern's
+    // implementaion, i.e., the users, and it can be dropped after we move the
+    // checks to the interface implementations.
```
Comment (jtuyls) [inline compiler/src/iree/compiler/Codegen/ExternalInterfaces/GPUEncodingExternalModels.cpp:593]: I am unsure about in which case this check can be dropped in the future? Do you want to support partial slices here?

Comment (hanhanW) [reply compiler/src/iree/compiler/Codegen/ExternalInterfaces/GPUEncodingExternalModels.cpp:593]: Yes, I plan to do it in a follow-up. I think it is very similar to https://github.com/iree-org/iree/pull/22360. The main difference between the two resolvers is that the padding resolver gets padded DispatchTensorType while the identity resolver gets the original DispatchTensorType. The offsets/strides/sizes and RankedTensorType types used in compute ops are the same.

**Review by jtuyls — APPROVED** [review]
LGTM
